[docs] Switch R docsite to custom gh runner

ebezzi opened the pull request:
R docsite generation now runs on GHA again using a runner with enough RAM.

ebezzi commented on .github/workflows/docsite-build-deploy.yml:
```diff
@@ -33,19 +33,39 @@ jobs:
         with:
           folder: docsite # The folder the action should deploy.
-          branch: gh-pages
+          branch: gh-pages-test
```
```suggestion
          branch: gh-pages
```

bkmartinjr commented on .github/workflows/docsite-build-deploy.yml:
```diff
@@ -9,7 +9,7 @@ permissions:
   build-and-deploy:
     concurrency: ci-${{ github.ref }}
-    runs-on: ubuntu-latest
+    runs-on: single-cell-1tb-runner
```
Naive question:  why do we need such a large instance type to generate docs?  Would `single-cell-8c64g-runner` suffice (I believe it uses the same base image, but on a cheaper/smaller instance type)

ebezzi replied:
It's actually a good question - I think historically some of the vignettes were very memory hungry, hence the 1T choice, but it's worth giving it a try with the 64G.

bkmartinjr replied:
I converted several to the small runner, to save $$

ebezzi replied:
Switched to `single-cell-8c64g-runner` and doing a test run right now.

ebezzi replied:
It worked! Leaving the change.

bkmartinjr reviewed (approved):
Couple of optional small items noted inline. At your discretion.

pablo-gar commented on api/r/cellxgene.census/vignettes/census_citation_generation.Rmd:
```diff
@@ -96,15 +96,15 @@ print(slice_datasets$citation)
-    obs_value_filter = "tissue == 'cardiac atrium'",
-    var_value_filter = "feature_name == 'MYBPC3'",
-    obs_column_names = c("dataset_id", "cell_type")
+  census = census,
```
Any particular reason we are removing indentation here?

mlin reviewed (commented):
@ebezzi This is great, thank you!!!! One thing, I think you can delete vignettes/README.md and update_docs.sh. Delighted to be rid of that stuff of course!

ebezzi commented:
> @ebezzi This is great, thank you!!!! One thing, I think you can delete vignettes/README.md and update_docs.sh. Delighted to be rid of that stuff of course!

Good point, I also updated `release_process.md` and removed the outdated steps.
